### ISSUE TITLE
[Multimedia collage] Product tile missing focus ring

### ISSUE THREAD

**Issue by svinkle** [Shopify/dawn#910]
## Issue summary

When navigating through page content with the keyboard, the Multimedia collage product tile does not feature a focus indicator. Sighted keyboard-only users may not know where the cursor has landed and lead to a frustrating user experience.

<details>
  <summary>Screenshots</summary>

  ![Gif of keyboard cursor moving through the multimedia collage section. The product tile does not show a visible focus ring.](https://user-images.githubusercontent.com/1392632/142447705-ec179b8f-74ce-414c-a9c5-602dd3b77bc9.gif)

</details>

#### HTML

```html
<div class="collage-card collage-product card-wrapper">
  <!-- … -->
  <h3 class="card-information__text h4">
    <a href="/products/derby-tier-backpack" class="full-unstyled-link">
      Derby Tier Backpack
    </a>
  </h3>
  <!-- … -->
</div>
```

## Steps to reproduce

1. Go to a Dawn theme store with any supported browser
2. <kbd>Tab</kbd> through a Multimedia Collage section

### Expected

- Links to feature a focus indicator.

### Actual

- Focus indicator is missing.

## Recommendation

Adjust the CSS to include the Dawn focus ring around the `div.collage-product` content container when its link receives focus.

#### CSS

```css
.collage-product:focus-within {
  outline: 0.2rem solid rgba(var(--color-foreground), 0.5);
  outline-offset: 0.3rem;
  box-shadow: 0 0 0 0.3rem rgb(var(--color-background)), 0 0 0.5rem 0.4rem rgba(var(--color-foreground), 0.3);
}
```

<!-- #### Diff

```diff
- line removed
+ line added
``` -->

## Specifications

- **Browser:** Any
- **Dawn version:** 2.4.0
- **Component:** Multimedia collage
- **WCAG Principal:** Operable
- **WCAG SC:** [2.4.7 Focus Visible](https://www.w3.org/WAI/WCAG21/Understanding/focus-visible.html)
- **Severity:** High
- **Effort:** Low
